Automattic/jetpack
# actions: Fix concurrency for gardening action

#### Changes proposed in this Pull Request:

The `github.ref` for a `pull_request_target` event is almost always
going to be "refs/heads/master". We want the concurrency to be based on
the PR branch, so adjust to use `github.event.pull_request.head.ref` for
`pull_request_target` events.

I think this may in fact be a bug, in that events on separate PRs in quick succession may cause one to be incorrectly cancelled. It's a bit hard to tell, but I think that https://github.com/Automattic/jetpack/actions/runs/1346243419 was cancelled because of https://github.com/Automattic/jetpack/actions/runs/1346243458 for example.

#### Jetpack product discussion


None

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Maybe create a PR to merge into this PR, do two label changes in quick succession, and see that one is cancelled and that it reports the correct branch name in the cancellation message?
* Maybe make two PRs to merge into this PR, do a label change on each in quick succession, and see that neither PR's event is cancelled.
